ale/drivers/dawn_drivers.py

```diff
@@ -199,3 +208,227 @@ def detector_center_line(self):
+            return self.ephemeris_times[-1] + (self.line_exposure_duration / 2)
+        except:
+            return spice.scs2e(self.spacecraft_id, self.label['IsisCube']['Instrument']['SpacecraftClockStopCount'])
+    
+    @property
+    def is_calibrated(self):
+        return self.label['IsisCube']['Archive']['ProcessingLevelId'] > 2
+
+    @property 
+    def has_articulation_kernel(self):
+        regex = re.compile('.*dawn_vir_[0-9]{9}_[0-9]{1}.BC')
+        return any([re.match(regex, i) for i in self.kernels])
+    
+    @property
+    def frame_chain(self):
+        if not hasattr(self, '_frame_chain'):
+            if self.has_articulation_kernel and self.is_calibrated:
+                self._frame_chain = super().frame_chain
+            else:
+                self._frame_chain = super().frame_chain
+                target_times = np.asarray([self.ephemeris_times[0], self.ephemeris_times[-1]])
+                sensor_times = np.array(self.ephemeris_times)
+
+                sensor_time_dependent_frames, sensor_constant_frames = self._frame_chain.frame_trace(self.sensor_frame_id, self.center_ephemeris_time, self._props.get('nadir', False))
+                target_time_dependent_frames, target_constant_frames = self._frame_chain.frame_trace(self.target_frame_id, self.center_ephemeris_time)
+
+                sensor_time_dependent_frames = list(zip(sensor_time_dependent_frames[:-1], sensor_time_dependent_frames[1:]))
+                constant_frames = list(zip(sensor_constant_frames[:-1], sensor_constant_frames[1:]))
+                target_time_dependent_frames = list(zip(target_time_dependent_frames[:-1], target_time_dependent_frames[1:]))
+                target_constant_frames = list(zip(target_constant_frames[:-1], target_constant_frames[1:]))
+
+                constant_frames.extend(target_constant_frames)
+                self._frame_chain.compute_time_dependent_rotiations(sensor_time_dependent_frames, sensor_times, 0)
+                self._frame_chain.compute_time_dependent_rotiations(target_time_dependent_frames, target_times, 0)
+
+                quats = np.zeros((len(self.optical_angles), 4))
+                for i, opt_angle in enumerate(self.optical_angles):
+                  
+                  rotation_matrix = spice.pxform("J2000", spice.frmnam(self.vir_zero_id), self.ephemeris_times[i])
+                  state_matrix = spice.rav2xf(rotation_matrix, [0, 0, 0])
+                  xform = spice.eul2xf([0, -opt_angle, 0, 0, 0, 0], 1, 2, 3)
+                  xform2 = spice.mxmg(xform, state_matrix)
+                  rot_mat, av = spice.xf2rav(xform2)
+                  
+                  quat_from_rotation = spice.m2q(rot_mat)
+                  quats[i,:3] = quat_from_rotation[1:]
+                  quats[i, 3] = quat_from_rotation[0]
+
+                  rotation = ConstantRotation(quats[i], spice.namfrm("J2000"), self.vir_zero_id)
+
+                  self._frame_chain.add_edge(rotation=rotation)
+
```

Comment:
I think the rosetta PR also does this  part, I would check there and compare. I know that the constant rotation should be a time dependent rotation though